smartsim/_core/mli/mli_schemas/response/response.capnp

```diff
@@ -0,0 +1,45 @@
+# BSD 2-Clause License
+
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+@0xa05dcb4444780705;
+
+using Tensors = import "../tensor/tensor.capnp";
+using ResponseAttributes = import "response_attributes/response_attributes.capnp";
+using DataRef = import "../data/data_references.capnp";
+
+struct Response {
+  status @0 :Int32;
+  statusMessage @1 :Text;
+  result :union {
+    keys @2 :List(DataRef.TensorKey);
+    data @3 :List(Tensors.Tensor);
```

Comment:
When a worker returns a tensor into a response, we can probably automate tensor shape and dtype conversion into this schema. But... We can't know the "order" value.

Should the response tensor metadata be specified in the original request so it can be specified by the user?

Reply:
Yeah, this is a great point that we should figure out.  I tend to like being more explicit about the return form (I think that's what you were implying here) since this is all internal tooling and not necessarily exposed to the user yet.  I think the less explicit option is that the return ``order`` is the same as the ``input`` order.  But again, I like the idea of being more explicit about it.

Reply:
Do you mean adding something like this to the `Request` schema?
```
outputOrder @0 :Enums.Order;
```

Or is this something we should add to the `RequestAttributes`? Something like:
```
struct TorchRequestAttributes {
  tensorType @0 :TorchTensorType;
  outputOrder @1 :Enums.Order;
}
```

While I do think it would be nice to shove extra info into the customAttributes, after typing it out I'm realizing I don't love that option because it might be less intuitive with `Request` in the title.